AWSPinpointTests/AWSPinpointNotificationManagerTests.m

```diff
@@ -0,0 +1,153 @@
+//
+// Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import "OCMock.h"
+#import "AWSTestUtility.h"
+#import "AWSPinpoint.h"
+#import "AWSPinpointNotificationManager.h"
+
+@interface AWSPinpointAnalyticsClient()
+@property (nonatomic, strong) NSDictionary* globalFeatureAttributes;
+- (void) setFeatureAttributes:(NSDictionary*) attributes;
+- (void) removeAllGlobalFeatureAttributes;
+@end
+
+@interface AWSPinpointNotificationManager()
+- (void)addPinpointMetadataForEvent:(AWSPinpointEvent *) event
+                   withNotification:(NSDictionary *) userInfo;
+- (void)addGlobalPinpointMetadataForNotification:(NSDictionary *) userInfo;
+- (NSString*)buildEventTypeFromUserInfo:(NSDictionary *) userInfo
+                     withPushActionType:(AWSPinpointPushActionType) pushActionType;
+@end
+
+@interface AWSPinpointConfiguration()
+@property (nonnull, strong) NSUserDefaults *userDefaults;
+@end
+
+@interface AWSPinpointNotificationManagerTests : XCTestCase
+@property (nonatomic, strong) AWSPinpoint *pinpoint;
+@property (nonatomic, strong) NSUserDefaults *userDefaults;
```

Comment:
Do we really need this need this as a property?  It looks like on line 60 this could just be:
```
configuration.userDefaults = [[NSUserDefaults alloc] initWithSuiteName:@"AWSPinpointNotificationManagerTests"];
```

Reply:
Will remove